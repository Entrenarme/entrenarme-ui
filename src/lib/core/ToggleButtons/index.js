// @flow
import * as React from 'react';
import styled from 'styled-components';
import JssProvider from 'react-jss/lib/JssProvider';
import { create } from 'jss';
import { createGenerateClassName, jssPreset } from '@material-ui/core/styles';

import Button from '../Button/index';
import colors from '../../helpers/colors';

const generateClassName = createGenerateClassName({
  productionPrefix: 'togglebuttons-eui',
});
const jss = create(jssPreset());

const MainContainer = styled.div`
  width: 100%;
`;

const LeftButtonContainer = styled.div`
  width: 50%;
  position: relative;
  display: inline-block;
`;

const LeftButton = styled(Button)`
  width: 100%;
  border-radius: 50px 0px 0px 50px !important;

  &:hover {
    background-color: ${colors.primary.active} !important;
  }
`;

const RightButton = styled(Button)`
  width: 50%;
  border-radius: 0px 50px 50px 0px !important;

  &:hover {
    background-color: ${colors.primary.active} !important;
  }
`;

const SeparationBar = styled.div`
  position: absolute;
  top: 22%;
  right: -1px;
  z-index: 20;

  background-color: #000;
  width: 2px;
  height: 55%;
  opacity: 0.15;
`;

type Props = {
  leftText: string,
  leftIcon: React.Node,
  leftOnclick: Function,
  rightText: string,
  rightIcon: React.Node,
  rightOnclick: Function,
};

const ToggleButton = ({
  leftText,
  leftIcon,
  leftOnclick,
  rightText,
  rightIcon,
  rightOnclick,
  ...rest
}: Props) => (
  <JssProvider jss={jss} generateClassName={generateClassName}>
    <MainContainer {...rest}>
      <LeftButtonContainer>
        <LeftButton icon={leftIcon} onClick={leftOnclick}>
          {leftText}
        </LeftButton>
        <SeparationBar />
      </LeftButtonContainer>
      <RightButton icon={rightIcon} onClick={rightOnclick}>
        {rightText}
      </RightButton>
    </MainContainer>
  </JssProvider>
);

export default ToggleButton;
