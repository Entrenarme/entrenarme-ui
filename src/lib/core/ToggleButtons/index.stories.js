// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCoffee } from '@fortawesome/free-solid-svg-icons';

import ToggleButtons from './index';

storiesOf('ToggleButtons', module).add(
  'Toggle buttons with icon',
  withInfo('ToggleButtons')(() => (
    <ToggleButtons
      leftText="Ver listado"
      rightText="Filtrar"
      leftIcon={<FontAwesomeIcon icon={faCoffee} />}
      rightIcon={<FontAwesomeIcon icon={faCoffee} />}
      leftOnclick={() => console.log('click left')}
      rightOnclick={() => console.log('click right')}
    />
  )),
);
