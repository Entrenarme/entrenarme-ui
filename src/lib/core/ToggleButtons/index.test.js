import * as React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCoffee } from '@fortawesome/free-solid-svg-icons';

import ToggleButtons from './index';

test('renders "ToggleButtons" with default config', () => {
  const { container } = render(
    <ToggleButtons
      leftText="Test left"
      rightText="Test right"
      leftIcon={<FontAwesomeIcon icon={faCoffee} />}
      rightIcon={<FontAwesomeIcon icon={faCoffee} />}
      leftOnclick={() => console.log('click left')}
      rightOnclick={() => console.log('click right')}
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});
