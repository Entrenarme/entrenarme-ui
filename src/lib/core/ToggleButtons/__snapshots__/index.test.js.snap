// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "ToggleButtons" with default config 1`] = `
<div
  class="sc-htpNat KtFsv"
>
  <div
    class="sc-bxivhb cvszCC"
  >
    <button
      tabindex="0"
      class="MuiButtonBase-root-23 MuiButton-root-1 MuiButton-flat-5 sc-ifAKCX ixzayO sc-bdVaJa kSjfNl"
      type="button"
      options="[object Object]"
    >
      <span
        class="MuiButton-label-2"
      >
        <span
          class="sc-bwzfXH czactv"
        >
          <svg
            aria-hidden="true"
            data-prefix="fas"
            data-icon="coffee"
            class="svg-inline--fa fa-coffee fa-w-20 "
            role="img"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 640 512"
          >
            <path
              fill="currentColor"
              d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
            />
          </svg>
        </span>
        <span>
          Test left
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-26"
      />
    </button>
    <div
      class="sc-bZQynM bfpLxY"
    />
  </div>
  <button
    tabindex="0"
    class="MuiButtonBase-root-23 MuiButton-root-1 MuiButton-flat-5 sc-EHOje hphWFf sc-bdVaJa kSjfNl"
    type="button"
    options="[object Object]"
  >
    <span
      class="MuiButton-label-2"
    >
      <span
        class="sc-bwzfXH czactv"
      >
        <svg
          aria-hidden="true"
          data-prefix="fas"
          data-icon="coffee"
          class="svg-inline--fa fa-coffee fa-w-20 "
          role="img"
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 640 512"
        >
          <path
            fill="currentColor"
            d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
          />
        </svg>
      </span>
      <span>
        Test right
      </span>
    </span>
    <span
      class="MuiTouchRipple-root-26"
    />
  </button>
</div>
`;
